src/Features/Core/Portable/Common/TaggedText.cs

```diff
@@ -27,15 +27,48 @@ namespace Microsoft.CodeAnalysis
         /// </summary>
         public string Text { get; }
 
+        /// <summary>
+        /// Gets the style(s) to apply to the text.
+        /// </summary>
+        internal TaggedTextStyle Style { get; }
+
+        /// <summary>
+        /// Gets the navigation target for the text, or <see langword="null"/> if the text does not have a navigation
+        /// target.
+        /// </summary>
+        internal string NavigationTarget { get; }
```

Comment:
i would prefer to pair this with a NavigationTargetKind, or have this be some sort of richer object that could prevent the consumer from having to do all the string-contortions to determine waht to do withthis.

Reply:
I'm going to file a follow-up issue to review this as part of moving it to public API.

Reply:
➡️ Filed follow-up issue #37500